include/dirac_quda.h

```diff
@@ -742,6 +765,59 @@ namespace quda {
     void reconstruct(ColorSpinorField &x, const ColorSpinorField &b, const QudaSolutionType) const;
   };
 
+  // Full Mobius EOFA
+  class DiracMobiusEofa : public DiracMobius
+  {
+
+  protected:
+    // The EOFA parameters
+    double m5inv_fac = 0.;
+    double sherman_morrison_fac = 0.;
+    double eofa_shift;
+    int eofa_pm;
+    double mq1;
+    double mq2;
+    double mq3;
+    double eofa_u[QUDA_MAX_DWF_LS];
+    double eofa_x[QUDA_MAX_DWF_LS];
+    double eofa_y[QUDA_MAX_DWF_LS];
+
+  public:
+    DiracMobiusEofa(const DiracParam &param);
+
+    void m5_eofa(ColorSpinorField &out, const ColorSpinorField &in) const;
+    void m5_eofa_xpay(ColorSpinorField &out, const ColorSpinorField &in, const ColorSpinorField &x, double a = -1.) const;
+
+    virtual void M(ColorSpinorField &out, const ColorSpinorField &in) const;
+    virtual void MdagM(ColorSpinorField &out, const ColorSpinorField &in) const;
+
+    virtual void prepare(ColorSpinorField *&src, ColorSpinorField *&sol, ColorSpinorField &x, ColorSpinorField &b,
+                         const QudaSolutionType) const;
+    virtual void reconstruct(ColorSpinorField &x, const ColorSpinorField &b, const QudaSolutionType) const;
+  };
+
+  // 4d Even-odd preconditioned Mobius domain wall with EOFA
+  class DiracMobiusPCEofa : public DiracMobiusEofa
```

Comment:
Naming convention: this should be `DiracMobiusEofaPC`, unless there's a specific reason we're interchanging the order (which we should note in the comment)